### DISCUSSION TITLE
[Forge 1.20.1] Support for Sky and Underground Villages (and Witch Villages from Villages & Pillages)

### DISCUSSION THREAD

**Discussion by vizthex123** [ejektaflex/Bountiful#277] [Mod Compat]
[Sky Villages](https://www.curseforge.com/minecraft/mc-mods/sky-villages-forge) and [Underground Villages](https://www.curseforge.com/minecraft/mc-mods/underground-villages) currently don't generate with bounty boards, but it'd be nice if they had support.

Also, a config option for [Witch Villages](https://www.curseforge.com/minecraft/mc-mods/villages-and-pillages) (off by default, perhaps?) would be nice.

And perhaps all mod compat could be disabled in the config? Would be a nice addition if pack makers only wanted certain villages to generate with boards.

**Comment by ejektaflex** [comment]
Changing this is generally a pain in the butt. It involves modifying the NBT of a block in the gazebo NBT file and creating an identical gazebo with a different "connector" that connects to the other jigsaws in their city's jigsaw system. If there was an easy way of doing it, I'd do it. As far as I know, there is not.

However, I did find this:
https://stackoverflow.com/questions/62040520/python-decoding-base64-nbt-gzip-what-is-it

It seems like we can unzip and decode an NBT file to get NBT data. If this NBT data is in text form (*not* binary), then I could possibly write a script to take our existing gazebo and make copies for each village we want compat with. If that works, then we could detect the presence of mods that add third party villages and inject into their jigsaw systems.

Something to note is that each mod that overwrites villages like this might be using different jigsaw weights.. as such, we might want to make the weights for this third party village stuff data driven.

Each NBT we generate a compat for will make our mod 1-2kb bigger - just a small consideration.

**Reply by vizthex123** [reply]
I have never once paid attention to the filesize of a mod lol (although I'd assume CF has an upload limit).

If you don't want to add it that'd be fine of course, but I wanted to suggest it anyway just in case.
